backend/api/Services/MissionSchedulingService.cs

```diff
@@ -463,56 +466,35 @@ IEnumerable<string> interruptedMissionRunIds
                     return;
                 }
 
-                var unfinishedTasks = missionRun
-                    .Tasks.Where(t =>
-                        !new List<Database.Models.TaskStatus>
-                        {
-                            Database.Models.TaskStatus.Successful,
-                            Database.Models.TaskStatus.Failed,
-                        }.Contains(t.Status)
-                    )
-                    .Select(t => new MissionTask(t))
-                    .ToList();
-
-                if (unfinishedTasks.Count == 0)
-                    continue;
-
-                var newMissionRun = new MissionRun
-                {
-                    Name = missionRun.Name,
-                    Robot = missionRun.Robot,
-                    MissionRunType = missionRun.MissionRunType,
-                    InstallationCode = missionRun.InspectionArea!.Installation.InstallationCode,
-                    InspectionArea = missionRun.InspectionArea,
-                    Status = MissionStatus.Pending,
-                    DesiredStartTime = missionRun.DesiredStartTime,
-                    Tasks = unfinishedTasks,
-                };
-
-                try
-                {
-                    await missionRunService.Create(
-                        newMissionRun,
-                        triggerCreatedMissionRunEvent: false
-                    );
-                    logger.LogInformation(
-                        "Interrupted mission run {MissionRunId} was rescheduled as new mission run {NewMissionRunId}",
-                        missionRun.Id,
-                        newMissionRun.Id
-                    );
-                }
-                catch (UnsupportedRobotCapabilityException)
-                {
-                    logger.LogError(
-                        "Unsupported robot capability detected when restarting interrupted missions for robot {robotName}. This should not happen.",
-                        missionRun.Robot.Name
-                    );
-                }
+                logger.LogInformation(
+                    "Moving interrupted mission run {MissionRunId} back to the queue",
+                    missionRun.Id
+                );
+                await missionRunService.UpdateMissionRunProperty(
+                    missionRun.Id,
+                    "Status",
+                    MissionStatus.Pending
+                );
+                _ = signalRService.SendMessageAsync(
+                    "Mission run created",
+                    missionRun?.InspectionArea?.Installation,
+                    missionRun != null ? new MissionRunResponse(missionRun) : null
+                );
             }
         }
 
         private async Task StartMissionRun(MissionRun queuedMissionRun, Robot robot)
         {
+            // Reset status reason in case this is a restarted mission run
```

Comment:
Could it make sense to have this be a separate function to describe the behavior enough so that a comment isn't necessary?

Reply:
When entering the function we don't know if its a restarted mission or not. One option is to create a IsRestartedMissionRun that checks if any of the tasks are not notStarted and then reset status reason